### PR TITLE
ROX-18155: pg generic store: regions

### PR DESCRIPTION
**Pull request by janisz**
## Description

A detailed explanation of the changes in your PR.

Feel free to remove this section if it is overkill for your PR, and the title of your PR is sufficiently descriptive.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

TODO(replace-me)
Use this space to explain how you tested your PR, or, if you didn't test it, why
you did not do so. Valid reasons include, for example, "CI is sufficient",
"No testable changes". Feel free to attach JSON snippets, curl commands,
screenshots.

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6766** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6766" target="_blank">Graphite</a> 
    * **PR #6767** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6767" target="_blank">Graphite</a> 
      * **PR #6768** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6768" target="_blank">Graphite</a> 
        * **PR #6769** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6769" target="_blank">Graphite</a> 
          * **PR #6770** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6770" target="_blank">Graphite</a> 
            * **PR #6771** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6771" target="_blank">Graphite</a> 
              * **PR #6772** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6772" target="_blank">Graphite</a> 
                * **PR #6773** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6773" target="_blank">Graphite</a> 
                  * **PR #6774** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6774" target="_blank">Graphite</a> 
                    * **PR #6775** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6775" target="_blank">Graphite</a> 
                      * **PR #6776** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6776" target="_blank">Graphite</a> 
                        * **PR #6777** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6777" target="_blank">Graphite</a>  👈
                          * **PR #6778** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/6778" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6777?utm_source=stack-comment).

File: pkg/search/postgres/store.go [hunk 1]
```diff
@@ -108,7 +108,7 @@ func (s *GenericStore[T, PT]) Exists(ctx context.Context, id string) (bool, erro
 		if ok, err := s.permissionChecker.ExistsAllowed(ctx); err != nil {
 			return false, err
 		} else if !ok {
-			return false, nil
+			return false, sac.ErrResourceAccessDenied
```
Comment (rhybrillou) [inline pkg/search/postgres/store.go:111]: It looks like this PR was missed by re-stack as it brings back behaviours for which changes were requested.
Please make sure it only has the minimal change set.

**Review by rhybrillou — CHANGES_REQUESTED** [review]
I think some of the changes that are showing here should not actually be part of this PR.
Please make sure that the only changes here are those linked to the use of regions in the store template and in the generated files.

**Review by rhybrillou — APPROVED** [review]

**Comment by janisz** [comment]
@janisz started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/6777).

**Pull request merged**